docs/01-app/03-api-reference/03-file-conventions/dynamic-routes.mdx

```diff
@@ -145,6 +145,270 @@ export default async function Page(props: PageProps<'/[locale]'>) {
+
+```tsx filename="app/blog/[slug]/page.tsx"
+import { Suspense } from 'react'
+
+export default async function Page({
+  params,
+}: {
+  params: Promise<{ slug: string }>
+}) {
+  return (
+    <div>
+      <h1>Blog Post</h1>
+      <Suspense fallback={<div>Loading...</div>}>
+        <Content slug={params.then((p) => p.slug)} />
+      </Suspense>
+    </div>
+  )
+}
+
+async function Content({ slug }: { slug: Promise<string> }) {
+  const { slug: resolvedSlug } = await slug
+  return <article>{/* Your content */}</article>
+}
+```
+
+#### With `generateStaticParams` + `<Suspense>`
+
+Good for frequently updating content where you don't need disk persistence.
+
+**Properties:**
+
+- **Build time**: Validates route, prerenders samples (e.g., `/1`, `/2`, `/3`)
+- **Prerendered params**: Instant - served from disk
+- **Runtime params**: Shell renders immediately, UI updates when content resolves
+- **Caching**: Memory only (cleared on server restart)
+
+```tsx filename="app/blog/[slug]/page.tsx"
+import { Suspense } from 'react'
+
+export async function generateStaticParams() {
+  return [{ slug: '1' }, { slug: '2' }, { slug: '3' }]
+}
+
+export default async function Page({
+  params,
+}: {
+  params: Promise<{ slug: string }>
+}) {
+  return (
+    <div>
+      <h1>Blog Post</h1>
+      <Suspense fallback={<div>Loading...</div>}>
+        <Content slug={params.then((p) => p.slug)} />
+      </Suspense>
+    </div>
+  )
+}
+
+async function Content({ slug }: { slug: Promise<string> }) {
+  const { slug: resolvedSlug } = await slug
+  return <article>{/* Your content */}</article>
+}
+```
+
+#### With `generateStaticParams` + `<Suspense>` + `use cache` on page
+
+Recommended for most use cases. Provides immediate UI for prerendered and previously visited routes, with disk persistence similar to ISR.
+
+**Properties:**
+
+- **Build time**: Validates route, prerenders samples
+- **Prerendered params**: Instant - served from disk
+- **Runtime params**: Shell renders immediately, UI updates when content resolves
+- **Caching**: Disk (persists across server restarts)
+
+```tsx filename="app/blog/[slug]/page.tsx"
+import { Suspense } from 'react'
+import { cacheLife } from 'next/cache'
+
+export async function generateStaticParams() {
+  return [{ slug: '1' }, { slug: '2' }, { slug: '3' }]
+}
+
+export default async function Page({
+  params,
+}: {
+  params: Promise<{ slug: string }>
+}) {
+  'use cache'
+  cacheLife('days')
+  return (
+    <div>
+      <h1>Blog Post</h1>
+      <Suspense fallback={<div>Loading...</div>}>
+        <Content slug={params.then((p) => p.slug)} />
+      </Suspense>
+    </div>
+  )
+}
+
+async function Content({ slug }: { slug: Promise<string> }) {
+  const { slug: resolvedSlug } = await slug
+  return <article>{/* Your content */}</article>
+}
+```
+
+#### With `generateStaticParams` + `use cache` on page (no Suspense)
+
+Full page waits but cached to disk. Good for less-frequently-changing content.
```

Comment:
Inverted case, you might have several thousands of items, but only want a subset to be always readily available, assuming there's tolerance for blocking navigations every now and then - I am almost just adding this for completeness sake tbh